MaxText/decode.py

```diff
@@ -133,15 +132,13 @@ def decode_loop(config, state=None):
 
   # Model and Optimizer definition
   model = Transformer(config, mesh = mesh)
-
-  tx = optax.adamw(
-    max_utils.create_learning_rate_schedule(config)
-  ) # TODO: we need an optax.GradientTransformation to form a TrainState, but we don't use it when decoding
-
-
   _, sp_tokenizer = create_data_iterator_with_tokenizer(config, mesh)
-
-  state, state_mesh_annotations = max_utils.setup_initial_state(model, tx, config, rng, mesh, checkpoint_manager)
+  state, state_mesh_annotations = max_utils.setup_decode_state(
+    model, config, rng, mesh, checkpoint_manager
+    )
```

Comment:
I think our linters might complain unless the ")" is moved one left here but I don't know!